api/services/v1alpha1/monitoring_types.go

```diff
@@ -188,6 +220,8 @@ type MonitoringCommonSpec struct {
 	Alerting *Alerting `json:"alerting,omitempty"`
 	// CollectorReplicas specifies the number of replicas in opentelemetry-collector, default is 2 if not set
 	CollectorReplicas int32 `json:"collectorReplicas,omitempty"`
+	// Resources defines the resource requests and limits for shared monitoring components
+	Resources *SharedResources `json:"resources,omitempty"`
```

Comment:
In metrics, we have `resources`, in traces we have `tempoResources` and at root of the Monitoring CR we have `resources` again. Maybe we coud name metrics and traces resources just `resources` and then at the root `sharedResources` or `collectorResources` or even just `resources` as it is done now?